File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/variables/types/ConstantType.java
package ch.rmy.android.http_shortcuts.variables.types;

import ch.rmy.android.http_shortcuts.realm.Controller;
import ch.rmy.android.http_shortcuts.realm.models.Variable;

public class ConstantType extends BaseVariableType implements SyncVariableType {

    @Override
    protected ConstantEditorFragment createEditorFragment() {
        return new ConstantEditorFragment();
    }

    @Override
    public String resolveValue(Controller controller, Variable variable) {
        return variable.getValue();
    }

}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/variables/types/ConstantType.kt
package ch.rmy.android.http_shortcuts.variables.types

import ch.rmy.android.http_shortcuts.realm.Controller
import ch.rmy.android.http_shortcuts.realm.models.Variable

internal class ConstantType : BaseVariableType(), SyncVariableType {

    override fun createEditorFragment(): ConstantEditorFragment {
        return ConstantEditorFragment()
    }

    override fun resolveValue(controller: Controller, variable: Variable): String {
        return variable.value!!
    }

}
